wp-includes/components/class-wp-service-worker-core-asset-caching-component.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * WP_Service_Worker_Core_Asset_Caching_Component class.
+ *
+ * @package PWA
+ */
+
+/**
+ * Use a network-first caching strategy for assets from core (in wp-includes).
+ *
+ * @since 0.6
+ */
+class WP_Service_Worker_Core_Asset_Caching_Component implements WP_Service_Worker_Component {
+
+	/**
+	 * Cache name.
+	 *
+	 * @var string
+	 */
+	const CACHE_NAME = 'core-assets';
+
+	/**
+	 * Adds the component functionality to the service worker.
+	 *
+	 * @since 0.6
+	 *
+	 * @param WP_Service_Worker_Scripts $scripts Instance to register service worker behavior with.
+	 */
+	public function serve( WP_Service_Worker_Scripts $scripts ) {
+		if ( is_admin() ) {
+			return;
+		}
+
+		$scripts->caching_routes()->register(
+			'^' . preg_quote( trailingslashit( includes_url() ), '/' ) . '.*',
```

Comment:
This should also include assets under `s.w.org`.

Reply:
Otherwise, all the emojis will show as broken images when offline.

Reply:
We'll need to request that responses from `s.w.org` be served with CORS headers, however.

Reply:
Requested here: https://meta.trac.wordpress.org/ticket/5477

Nevertheless, the emoji responses from `s.w.org` is are expiring in  Thu, 31 Dec 2037 23:55:55 GMT. So we should actually be able to rely on the normal browser cache for such assets and not have to explicitly cache them in the service worker.

Reply:
Also, it would be problematic to include emoji in the same cache bucket as other core assets, since it is relatively an unbounded number of assets. The assets in core are a fixed number, but the number of emoji are vast, hence why they are distributed via a CDN. So I don't think they should be included in the end.